### PR TITLE
[MRG] Bugfix Decoder Dummy Classifier

### PR DESCRIPTION
**Pull request by tbng**
* Closes gh-2768 
* Closes gh-2767
* Continuation of PR gh-2769

Multiple changes for Decoder.Dummyclassifier

* Adding better test case for Dummy Classifier: now taken into account of the affect of distribution of the label on predicted value  
* Change default of the Dummy Classifier to `strategy='stratified'`
* Simplify calculation of prediction value

### CODE REVIEW

File: nilearn/decoding/tests/test_decoder.py [hunk 1]
```diff
@@ -184,51 +184,91 @@ def test_decoder_binary_classification():
...
+    y = np.zeros(n_samples)
+    y[:int(proportion * n_samples)] = 1.0
+
     model = Decoder(estimator='dummy_classifier', mask=mask)
     model.fit(X, y)
     y_pred = model.predict(X)
-    assert model.scoring == "roc_auc"
-    assert model.score(X, y) == 0.5
-    assert accuracy_score(y, y_pred) == 0.5
+    assert np.sum(y_pred == 1.0) / n_samples - proportion < 0.5
```
Comment (bthirion) [inline nilearn/decoding/tests/test_decoder.py:230]: Isn't this bound  bit loose ?

Comment (tbng) [reply nilearn/decoding/tests/test_decoder.py:230]: Yes, potentially so, but because the Dummy Classifier use numpy.random.multinomial to generate prediction label based on the proportion of label in training set so there would be a fair amount of randomness.

I think one way to decrease the bound here and still guarantee the test would not fail due to randomness is to increase the sample size (maybe 3x-4x), wdyt?

Comment (bthirion) [reply nilearn/decoding/tests/test_decoder.py:230]: Yes, if it's not too time consuming.

File: nilearn/decoding/decoder.py [hunk 2]
```diff
@@ -57,9 +55,9 @@
     ridge_regressor=RidgeCV(),
     ridge=RidgeCV(),
     svr=SVR(kernel='linear', max_iter=1e4),
-    dummy_classifier = DummyClassifier(strategy='prior',
-                                       random_state=0),
-    dummy_regressor = DummyRegressor(strategy='mean'),
+    dummy_classifier=DummyClassifier(strategy='stratified',
```
Comment (NicolasGensollen) [inline nilearn/decoding/decoder.py:58]: Could you add spaces around "="?

File: nilearn/decoding/decoder.py [hunk 3]
```diff
@@ -57,9 +55,9 @@
...
+                                     random_state=0),
+    dummy_regressor=DummyRegressor(strategy='mean'),
```
Comment (NicolasGensollen) [inline nilearn/decoding/decoder.py:60]: same

**Review by NicolasGensollen — COMMENTED** [review]
This is looking good, thanks @tbng !
I tried it locally and the dummy estimators now behave as expected.
Let us know when you think this is ready (title still says [WIP]).

File: nilearn/decoding/decoder.py (hunk 2 above)
Comment (tbng) [reply nilearn/decoding/decoder.py:58]: Sorry this is automatically corrected by my editor. Actually it's a bit weird because the above estimator definition does not have spaces around "=".  WDYT?

Comment (NicolasGensollen) [reply nilearn/decoding/decoder.py:58]: Sorry, I misunderstood what you were saying @tbng, thanks for clarifying! 
Indeed, this convention is mostly respected across nilearn, but there are still a few exceptions here and there...

**Review by NicolasGensollen — COMMENTED** [review]
This looks good to me, thanks @tbng !
Could you add a whatsnew entry?

**Comment by NicolasGensollen** [comment]
Hi @tbng 
I think this was ready to merge minus the missing whatsnew entry. 
Could you also merge master such that we get the latest's fixes? 
Thx!

**Comment by tbng** [comment]
> Hi @tbng
> I think this was ready to merge minus the missing whatsnew entry.
> Could you also merge master such that we get the latest's fixes?
> Thx!

Sorry I was neglecting a bit for finishing this PR, but I just added the entry

**Comment by bthirion** [comment]
Can you solve the conflict ?Thx

**Review by NicolasGensollen — APPROVED** [review]
LGTM, thanks!

**Comment by NicolasGensollen** [comment]
Merging, thanks!
